doc/users-guide/users-guide-pktio.adoc

```diff
@@ -136,330 +61,26 @@ struct, in turn, specifies the input and output *modes* of the PktIO.
- * ODP_PROTO_LAYER_L3 or higher. Similarly, L4 level checksum checking
- * may be enabled only with parsing level ODP_PROTO_LAYER_L4 or higher.
- *
- * Whether checksum checking was done and whether a checksum was correct
- * can be queried for each received packet with odp_packet_l3_chksum_status()
- * and odp_packet_l4_chksum_status().
- */
-typedef union odp_pktin_config_opt_t {
-	/** Option flags */
-	struct {
-		/** Timestamp all packets on packet input */
-		uint64_t ts_all        : 1;
-
-		/** Timestamp (at least) IEEE1588 / PTP packets
-		  * on packet input */
-		uint64_t ts_ptp        : 1;
-
-		/** Check IPv4 header checksum on packet input */
-		uint64_t ipv4_chksum   : 1;
-
-		/** Check UDP checksum on packet input */
-		uint64_t udp_chksum    : 1;
-
-		/** Check TCP checksum on packet input */
-		uint64_t tcp_chksum    : 1;
-
-		/** Check SCTP checksum on packet input */
-		uint64_t sctp_chksum   : 1;
-
-		/** Drop packets with an IPv4 error on packet input */
-		uint64_t drop_ipv4_err : 1;
-
-		/** Drop packets with an IPv6 error on packet input */
-		uint64_t drop_ipv6_err : 1;
-
-		/** Drop packets with a UDP error on packet input */
-		uint64_t drop_udp_err  : 1;
-
-		/** Drop packets with a TCP error on packet input */
-		uint64_t drop_tcp_err  : 1;
-
-		/** Drop packets with a SCTP error on packet input */
-		uint64_t drop_sctp_err : 1;
-
-	} bit;
-
-	/** All bits of the bit field structure
-	  *
-	  * This field can be used to set/clear all flags, or bitwise
-	  * operations over the entire structure. */
-	uint64_t all_bits;
-} odp_pktin_config_opt_t;
------
-These are used to control packet timestamping as well as default packet checksum
-verification processing.
+controls the RX processing to be performed on received packets. For example,
+`odp_pktin_config_opt_t` includes options for controlling packet timestamping as
+well as default packet checksum verification processing.
 
 ==== PktIO Parsing Configuration
 For RX processing, packets may also be parsed automatically as part of
-receipt as controlled by the `odp_pktio_parser_config_t` struct:
-[source,c]
------
-/**
- * Parser configuration
- */
-typedef struct odp_pktio_parser_config_t {
-	/** Protocol parsing level in packet input
-	  *
-	  * Application requires that protocol headers in a packet are checked
-	  * up to this layer during packet input. Use ODP_PROTO_LAYER_ALL for
-	  * all layers. Packet metadata for this and all preceding layers are
-	  * set. In addition, offset (and pointer) to the next layer is set.
-	  * Other layer/protocol specific metadata have undefined values.
-	  *
-	  * The default value is ODP_PROTO_LAYER_ALL. */
-	odp_proto_layer_t layer;
-
-} odp_pktio_parser_config_t;
------
-Note that parsing is automatically done whenever classification is enabled
-for an RX interface (see below).
+receipt as controlled by the `odp_pktio_parser_config_t` struct. Note that
+parsing is automatically done whenever classification is enabled for an RX
```

Comment:
> parsing is automatically done whenever classification is enabled

This statement caught my eye even though it is not changed by this commit. I wonder if enabling of parsing is really guaranteed by the API in this case as I did not find it in the API text. Or perhaps this is just the way the linux-gen ODP implementation works?

Reply:
Indeed the API doesn't mention this, so removed the sentence.